bin/live/pycbc_live_collate_triggers

```diff
@@ -0,0 +1,306 @@
+    # Convert dates to datetime to get the days array
+    start_date = datetime.strptime(args.start_date, '%Y-%m-%d').date()
+    num_days = timedelta(days=args.num_days - 1)
+    end_date = start_date + num_days
+
+    # Create the days array
+    days = []
+    delta = timedelta(days=1)
+
+    # Loop through the days and add each date to the list
+    current_date = start_date
+    while current_date <= end_date:
+        days.append(current_date)
+        current_date += delta
+
+    # Convert to strings in a list
+    days = [str(date).replace('-', '_') for date in days]
+    num_days = str(len(days))
+
+    trigger_files = [
+        os.path.join(args.trigger_dir, day, trigger_file)
+        for day in days
+        for trigger_file in os.listdir(os.path.join(args.trigger_dir, day))
+    ]
+
+if args.trigger_file_method == 'gps-start-end-time':
+    assert args.gps_start_time, 'Please provide a GPS start time.'
+    assert args.gps_end_time, 'Please provide a GPS end time.'
+    assert args.trigger_dir, 'Please provide a directory containing trigger files.'
+
+    trigger_files = [
+        l for l in glob.glob(args.trigger_dir + '/*/*-Live-*.hdf', recursive=True)
+        if float(l.split('/')[-1].split("-")[2]) > args.gps_start_time and float(l.split('/')[-1].split("-")[2]) < args.gps_end_time
+    ]
+
+start = timeit.default_timer()
+logging.info(f" {len(trigger_files)} files found")
+
+if args.output_trigger_file_list:
+    logging.info(' Writing list of trigger files to: ' + args.output_trigger_file_list)
+    with open(args.output_trigger_file_list, 'w') as f:
+        for item in trigger_files:
+            f.write("%s\n" % item)
+
+# Record start and end gpstime for the output file
+#  This assumes the trigger files are sorted chronologically
+start_gpstime = float(trigger_files[0].split('/')[-1].split("-")[2])
+end_gpstime = float(trigger_files[0].split('/')[-1].split("-")[2])
+
+###########################
+# COLLATE THE TRIGGER FILES
+###########################
+
+if args.output_file_name:
+    output_file = args.output_dir + args.output_file_name
+else:
+    ifo_string = "".join(args.ifos)
+    print(ifo_string)
+    output_file = args.output_dir + f'{ifo_string}-Live-' + start_gpstime + '-' + end_gpstime + '.hdf'
+
+logging.info(f" Creating a file at: {output_file}")
+with h5py.File(output_file, 'a') as destination:
+    # Create the ifo groups in the trigger file
+    for ifo in args.ifos:
+        if ifo not in destination:
+            destination.create_group(ifo)
+
+    # The user must give the ifos in the same order as the trigger files
+    #  For PyCBC Live it's always H1L1 or H1L1V1?
+    ifo_prefix = "".join(args.ifos)
+
+    file_count = 0
+    for source_file in trigger_files:
+        file_count += 1
+        trigger_file = os.path.basename(source_file)
+        if not (trigger_file.endswith('.hdf') and trigger_file.startswith(f'{ifo_prefix}-Live')):
+            continue
```

Comment:
Can you do this part during the trigger_files calculation earlier? By doing it this way your progress logging is misleading

Reply:
Good spot, I missed this when squishing the two scripts together. This will record every file (pngs, xmls, etc.) as a trigger file until checked later. I will move this check. Thanks!

Reply:
Okay, I've moved it out and added the ifo permutation part